### PR TITLE
Disable identity for local mounts

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

File: vault/request_handling.go [hunk 1]
```diff
@@ -496,7 +496,9 @@ func (c *Core) handleLoginRequest(ctx context.Context, req *logical.Request) (re
 		var entity *identity.Entity
 		auth = resp.Auth
 
-		if auth.Alias != nil {
+		mEntry := c.router.MatchingMountEntry(req.Path)
+
+		if auth.Alias != nil && mEntry != nil && !mEntry.Local {
```
Comment (jefferai) [inline vault/request_handling.go]: You're missing the auth/token/renew path earlier on, which can also end up updating memberships.

File: vault/identity_store.go [hunk 2]
```diff
@@ -329,6 +329,10 @@ func (i *IdentityStore) CreateOrFetchEntity(alias *logical.Alias) (*identity.Ent
 		return nil, fmt.Errorf("invalid mount accessor %q", alias.MountAccessor)
 	}
 
+	if mountValidationResp.MountLocal {
```
Comment (jefferai) [inline vault/identity_store.go]: Shouldn't this happen later so that existing values are returned?

File: vault/request_handling.go (hunk 1 above)
Comment (vishalnayak) [reply vault/request_handling.go]: I left the renewal flow as it is to allow it to work if the tokens were created before this fix. The thought I had was to only disallow entity attachment for the newly created tokens.

File: vault/identity_store.go (hunk 2 above)
Comment (vishalnayak) [reply vault/identity_store.go]: It wouldn't matter. This function will not be called for the new tokens.

**Pull request merged**